packages/Recognizers/ConversationLanguageUnderstanding/dotnet/CluApplication.cs

```diff
@@ -0,0 +1,97 @@
+﻿using System;
+
+namespace Microsoft.Bot.Components.Recognizers.CLURecognizer
+{
+    /// <summary>
+    /// Data describing a CLU application.
+    /// </summary>
+    public class CluApplication
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CluApplication"/> class.
+        /// </summary>
+        /// <param name="projectName">CLU project name.</param>
+        /// <param name="endpointKey">CLU subscription or endpoint key.</param>
+        /// <param name="endpoint">CLU endpoint to use.</param>
+        /// <param name="deploymentName">CLU deployment name.</param>
+        public CluApplication(string projectName, string endpointKey, string endpoint, string deploymentName)
+            : this((projectName, endpointKey, endpoint, deploymentName))
+        {
+        }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="CluApplication"/> class.
+        /// For unit tests only.
+        /// </summary>
+        protected CluApplication()
+        {
+        }
+
+        private CluApplication(ValueTuple<string, string, string, string> props)
```

Comment:
Why does this exist?  Why not use the typed constructor only?

Reply:
Never mind.  I see it's a copy of the existing recognizer.  Though I don't understand that either.  ;-)  We can go with this.